### PR TITLE
Adding Datadog agent to Heroku build

### PR DESCRIPTION
**Pull request by DemiAHMS**

### CODE REVIEW

File: bin/compile [hunk 1]
```diff
@@ -45,6 +45,10 @@ fi
 $CACHE_DIR/awscli/awscli-bundle/install -b ~/bin/aws
 export PATH=~/bin:$PATH
 
+### Install Datadog agent
+DD_AGENT_MAJOR_VERSION=7 DD_API_KEY=0ff0a65287d17527a4c6395df86a4046 DD_SITE="datadoghq.com" bash -c "$(curl -L https://s3.amazonaws.com/dd-agent/scripts/install_script.sh)"
```
Comment (sbdchd) [inline bin/compile:49]: is this API key supposed job e secret?

Comment (sbdchd) [reply bin/compile:49]: I think if we add it via `heroku config` the env should be available in the install script's env

Comment (DemiAHMS) [reply bin/compile:49]: It shouldn't be public-on-the-internet, but it doesn't need to be secret-and-limited-visibility

Comment (DemiAHMS) [reply bin/compile:49]: Although I just realized those are already in the Heroku configs

Comment (sbdchd) [reply bin/compile:49]: This repo is public FYI

Comment (DemiAHMS) [reply bin/compile]: Squashed and forced to remove them

File: bin/compile [hunk 2]
```diff
@@ -47,6 +47,7 @@ export PATH=~/bin:$PATH
 
 ### Install Datadog agent
 bash -c "$(curl -L https://s3.amazonaws.com/dd-agent/scripts/install_script.sh)"
+python3 pip install ddtrace             # For Datadog APM
```
Comment (sbdchd) [inline bin/compile:50]: do we want to pin the version via `==`?

Comment (DemiAHMS) [reply bin/compile:50]: Probably not - pinning the version just adds more maintenance overhead for us, or we end up with an ancient version years from now

Comment (sbdchd) [reply bin/compile:50]: I think the advantage is we avoid the surprise of a new version breaking our build

Comment (DemiAHMS) [reply bin/compile:50]: I think the risk calculus of "low possibility of broken build in the future" vs "definite security vulnerability in the future from pinned, obsolete package" only comes out one way.
If a new version _does_ break our build, that is something we should look into; that is, we want it to break our build and force us to figure out the incompatibility.

File: bin/compile (hunk 1 above)
Comment (chdsbd) [reply bin/compile]: need to regenerate that key, it's still visible

Comment (DemiAHMS) [reply bin/compile]: fixed

**Pull request closed without merging**